lib/sql.js

```diff
@@ -421,17 +421,13 @@ SQLConnector.prototype.propertyName = function(model, column) {
 };
 
 /**
- * Get the id column name
+ * Get the id column name.
  * @param {String} model The model name
  * @returns {String} The id column name
  */
 SQLConnector.prototype.idColumn = function(model) {
-  let name = this.getDataSource(model).idColumnName(model);
-  const dbName = this.dbName;
-  if (typeof dbName === 'function') {
-    name = dbName(name);
-  }
-  return name;
+  const idName = this.getDataSource(model).getModelDefinition(model).idName();
```

Comment:
I have little troubles trying to understand how exactly was the old code working, how is the new code working, and what are the important differences.

For anybody else reviewing these changes, here are few pointers to better understand how is `this.getDataSource(model).idColumnName(model)` implemented under the hood:

https://github.com/strongloop/loopback-datasource-juggler/blob/36303b3b63932478cb429b5c5af6fec7f09b602c/lib/datasource.js#L2104-L2106
```js
DataSource.prototype.idColumnName = function(modelName) {
  return this.getModelDefinition(modelName).idColumnName(this.connector.name);
};
```

https://github.com/strongloop/loopback-datasource-juggler/blob/36303b3b63932478cb429b5c5af6fec7f09b602c/lib/model-definition.js#L162-L164
```js
ModelDefinition.prototype.idColumnName = function(connectorType) {
  return this.columnName(connectorType, this.idName());
};
```

https://github.com/strongloop/loopback-datasource-juggler/blob/36303b3b63932478cb429b5c5af6fec7f09b602c/lib/model-definition.js#L78-L89
```js
ModelDefinition.prototype.columnName = function(connectorType, propertyName) {
  if (!propertyName) {
    return propertyName;
  }
  this.build();
  const property = this.properties[propertyName];
  if (property && property[connectorType]) {
    return property[connectorType].column || property[connectorType].columnName || propertyName;
  } else {
    return propertyName;
  }
};
```

Essentially, the original implementation performs the following steps:
- inside juggler:
  - (1) calls `ModelDefinition.prototype.idName()` to find the property name of the primary key
  - (2) calls `ModelDefinition.prototype.columnName()` to map the property name to the column name
  - (3) inside `columnName()`, we try to use connector-specific configuration if it was provided in property metadata
- in this connector:
  - (4) call `dbName` to apply naming convention provided by the connector

In the new implementation, we keep the first step as-is, but replace steps 2-4 with a call of `this.column()` (see [the source code](https://github.com/strongloop/loopback-connector/blob/271371f869c5da6d17b0f68c889713f743564d75/lib/sql.js#L375-L395)). Under the hood, that means:
1. try to use connector-specific configuration if it was provided in property metadata (e.g. `{postgresql: {column: 'foo'}}`
2. try to use `name` field if provided in property metadata (e.g. `{name: 'foo}`)
3. only as the last resort, try to apply `dbname()` transformation

Based on the above analysis, I think the new implementation makes sense :+1:

I would like us to consider the bigger picture though and open a follow-up story and/or pull request(s) to address the root cause in all connectors, not only SQL ones.

- Considering that `DataSource.prototype.idColumnName` is effectively broken, I think we should mark it as deprecated and advise connectors to use `ModelDefinition.prototype.idName()` and `SqlConnector.prototype.column()` API instead. IIRC, DataSource methods are legacy from times before we introduced `loopback-connector` helpers (base connector classes). They have been superseded by helper methods provided by `Connector`/`SqlConnector`, but we never cleaned up this area of our codebase.
- IMO, NoSQL connectors (MongoDB, CouchDB/Cloudant) should use the same logic for mapping properties to column names (document property names) as the SQL connectors do. To allow that, we should push `column()` method to base `Connector` class, preferably finding a better name. E.g. `getColumnName`, or even better a name that's not specific to SQL columns - maybe `getFieldName` or `getPersistedName`.
- Since the new behavior will be no longer specific to SQL connectors, we should push the implementation of `SQLConnector.prototype.idColumn` to `Connector.prototype.idColumn`.
- We should update all connectors maintained by us to stop using `DataSource.prototype.idColumnName` and use the new Connector API instead.